tf2onnx/onnx_opset/tensor.py

```diff
@@ -1072,11 +1079,21 @@ def version_1(cls, ctx, node, **kwargs):
 
         attr = {"axes": slice_axis, "ends": ends, "starts": starts}
         inputs_map = {"data": trans2.output[0], **attr}
-        kwargs = {**inputs_map, "outputs": node.output}
         dtypes = [ctx.get_dtype(node.output[0])]
-        shapes = [ctx.get_shape(node.output[0])]
-        ctx.remove_node(node.name)
-        GraphBuilder(ctx).make_slice(kwargs, name=node.name, dtypes=dtypes, shapes=shapes)
+        shapes = ctx.get_shape(node.output[0])
+
+        if len(ctx.get_shape(input_tensor.output[0])) == 3:
+            # add a reshape op to convert output into 3d
+            kwargs = {**inputs_map}
+            ctx.remove_node(node.name)
+            slice1 = GraphBuilder(ctx).make_slice(kwargs)
+            shape_name = utils.make_name(node.name)
+            ctx.make_const(shape_name, np.array(shapes, dtype=np.int64))
+            ctx.make_node("Reshape", [slice1, shape_name], outputs=node.output, name=node.name, dtypes=dtypes)
```

Comment:
if squeeze used, do we need to make_const?

Reply:
for sometimes, the shapes will be [N, -1, -1, -1], and reshape's input can accept one -1 at most